README.md

```diff
@@ -7,6 +7,29 @@ An opinionated static site generator written in Rust.
 Documentation is available on [its site](https://www.getgutenberg.io/documentation/getting-started/installation/) or
 in the `docs/content` folder of the repository.
 
+## Comparisons with other static site generators
+
+|                          | Gutenberg | Cobalt | Hugo | Pelican |
+|--------------------------|-----------|--------|------|---------|
+| Single binary            |     ✔     |    ✔   |   ✔  |    ✕    |
+| Language                 |    Rust   |  Rust  |  Go  |  Python |
+| Syntax highlighting      |     ✔     |    ✔   |   ✔  |    ✔    |
+| Sass compilation         |     ✔     |    ✔   |   ✕  |    ✔    |
+| Assets co-location       |     ✔     |    ✕   |   ✔  |    ✔    |
```

Comment:
Being able to have a folder with images/other files next to the .md like https://github.com/Keats/gutenberg/tree/master/test_site/content/posts/with-assets

Reply:
This is exclusively how cobalt works.

I didn't even realize gutenberg supported this because I saw reference to a [`static` folder](https://www.getgutenberg.io/documentation/getting-started/directory-structure/) in the docs but saw no other reference to assets being able to sit next to everything else.

Reply:
Ah I missed that, I'll fix it.

Reply:
How does it work in Cobalt if you each .md has a `screenshot.png` file with it? Can you put the .md in a folder or do you have to have a unique asset name across the site?

Reply:
I'm not sure of the question which makes me wonder if there is a feature implicit in the question that gutenberg has that cobalt doesn't.

Assets do not need to have unique names.  They are copied straight over from the site source to the destination.

Reply:
Have a look at https://github.com/Keats/gutenberg/tree/master/docs/content/themes

This is generated from a git repository and they are all called `screenshot.png`. You obviously can't put several `screenshot.png` in the same folder so you have 2 choices:

- rename them to have unique names
- put them in a subfolder (what I am doing there)

If you put them in a subfolder, you might run into a destination path issue depending on how cobalt is setup up. In the above I want the path to be `themes/after-dark/` for example, not `themes/after-dark/index` without having to edit the path manually in my front-matter. What would be the paths for those pages in Cobalt?

Reply:
Ok, so the concern is when having files assets with pages that use `path/` instead of `path.html`.

`themes/after-dark/index.md` would be written to disk as `themes/after-dark/index.html`.

cobalts default permalink is `/{{parent}}/{{name}}{{ext}}` and since we are appending the extension, we don't append a `/index.html` to it.

If someone overrides the permalink to not include an extension, then they will run into the problem you mentioned.